### DISCUSSION TITLE
Filament | FileUpload Issue | Waiting for size | Loading

### DISCUSSION THREAD

**Discussion by markwillow** [spatie/laravel-google-cloud-storage#87] [Q&A]
Hi Everyone,

I hope you are all well. I have a problem and need your help on this. I am using Filament PHP and GCP to store my data. Here's my filesystem.php configuration:

'gcs-media' => [
            'driver' => 'gcs',
            'project_id' => env('GOOGLE_CLOUD_PROJECT_ID', 'project_id'),
            'bucket' => env('GOOGLE_CLOUD_MEDIA_STORAGE_BUCKET', 'https://test.com'),
            'visibility' => 'public',
            'bucket_url' => env('GOOGLE_CLOUD_MEDIA_BUCKET_URL'),
            'storage_api_uri' => env('GOOGLE_CLOUD_MEDIA_BUCKET_URL'),
            'visibility_handler' => \League\Flysystem\GoogleCloudStorage\UniformBucketLevelAccessVisibility::class,
            'key_file' => [
                'type' => env('GOOGLE_CLOUD_MEDIA_ACCOUNT_TYPE'),
                'private_key_id' => env('GOOGLE_CLOUD_MEDIA_PRIVATE_KEY_ID'),
                'private_key' => env('GOOGLE_CLOUD_MEDIA_PRIVATE_KEY'),
                'client_email' => env('GOOGLE_CLOUD_MEDIA_CLIENT_EMAIL'),
                'client_id' => env('GOOGLE_CLOUD_MEDIA_CLIENT_ID'),
                'auth_uri' => env('GOOGLE_CLOUD_MEDIA_AUTH_URI'),
                'token_uri' => env('GOOGLE_CLOUD_MEDIA_TOKEN_URI'),
                'auth_provider_x509_cert_url' => env('GOOGLE_CLOUD_MEDIA_AUTH_PROVIDER_CERT_URL'),
                'client_x509_cert_url' => env('GOOGLE_CLOUD_MEDIA_CLIENT_CERT_URL'),
            ],
            'throw' => true,
        ],

Here's my resource file too:

FileUpload::make('attachments')
                    ->columnSpan(2)
                    ->label('Attachments')
                    ->multiple()
                    ->previewable()
                    ->reorderable()
                    ->downloadable()
                    ->openable()
                    ->directory('attachments')
                    ->storeFileNamesIn('original_filenames')
                    ->disabled($user->cannot('admin:comment:view_attachments'))
                    ->hidden($user->cannot('admin:comment:edit_attachments')),

After uploading the image, there is no preview. So basically, it keeps on loading. The open and download features are working fine. But I had some issues with the preview after uploading it.

Hope you can light us with this.

**Comment by beelalamin** [comment]
Please refer to the below thread:
https://v2.filamentphp.com/tricks/file-upload-previews-not-loading
